src/components/ActiveWalletSection/index.tsx

```diff
@@ -0,0 +1,269 @@
+          "Your connected wallet doesn't match any Gameplay wallet tracked for this Quest. To track progress with this wallet, add it as a Gameplay Wallet below by setting it."
+        )}
+      </span>{' '}
+      <span className={cn('body-sm', styles.verifyText)}>
+        {t(
+          'wallet.verify.message',
+          'You only need to verify each address once and can switch freely at any time.'
+        )}
+      </span>
+    </InfoAlert>
+  )
+
+  const setButton = (
+    <Button
+      disabled={isPending}
+      type="secondaryGradient"
+      className={styles.setButton}
+      onClick={() => setActiveWalletMutation()}
+    >
+      {isPending ? (
+        <LoadingSpinner className={styles.loadingSpinner} />
+      ) : (
+        t('wallet.action.set', 'Set')
+      )}
+    </Button>
+  )
+
+  const hasNoWallets = !connectedWallet && !activeWallet
+  const hasOnlyConnectedWallet = connectedWallet && !activeWallet
+  const hasOnlyActiveWallet = activeWallet && !connectedWallet
+  const hasMatchingWallets =
+    Boolean(activeWallet && connectedWallet) && activeWallet === connectedWallet
+  const hasDifferentWallets =
+    Boolean(activeWallet && connectedWallet) && !hasMatchingWallets
+
+  let content = null
+
+  if (hasNoWallets) {
+    content = (
+      <>
+        <Alert
+          message={t(
+            'wallet.noWallet.message',
+            'No wallet connected. Connect wallet to track Quest progress.'
+          )}
+          variant="warning"
+        />
+        <InputLikeContainer
+          title={t('wallet.active.title', 'Active Gameplay Wallet')}
+        >
+          <InputLikeBox className={styles.noWallet}>
+            {t('wallet.noWallet.status', 'No wallet connected')}
+          </InputLikeBox>
+        </InputLikeContainer>
+      </>
+    )
+  }
+
+  if (hasOnlyConnectedWallet) {
+    content = (
+      <>
+        {onlyConnectedWallet}
+        <InputLikeContainer
+          title={t('wallet.connected.title', 'Connected Wallet')}
+        >
+          <div className={styles.setActiveWalletContainer}>
+            <InputLikeBox className={styles.setConnectedWalletInput}>
+              {truncateEthAddress(connectedWallet ?? '')}
+            </InputLikeBox>
+            {setButton}
+          </div>
+        </InputLikeContainer>
+      </>
+    )
+  }
+
+  if (hasOnlyActiveWallet) {
+    content = (
+      <InputLikeContainer
+        title={t('wallet.active.title', 'Active Gameplay Wallet')}
+      >
+        <InputLikeBox className={styles.activeWallet}>
+          {truncateEthAddress(activeWallet)}
+        </InputLikeBox>
+      </InputLikeContainer>
+    )
+  }
+
+  if (hasMatchingWallets) {
+    content = (
+      <InputLikeContainer title={t('wallet.connected.title', 'Active Wallet')}>
+        <InputLikeBox className={styles.activeWallet}>
+          {truncateEthAddress(activeWallet ?? '')}
+        </InputLikeBox>
+      </InputLikeContainer>
+    )
+  }
+
+  if (hasDifferentWallets) {
```

Comment:
should this condition be `hasDifferentWallets && hasNotSignedWithDifferentWallet`?

i.e. if I 
1. connect 0x123
2. set gameplay wallet to 0x123
3. switch to 0x444
4. set gameplay wallet to 0x444
5. switch back to 0x123

It shouldn't ask me to sign again right?

Reply:
good call. I need to add this UX